Lib/test/_test_multiprocessing.py

```diff
@@ -3903,6 +3890,46 @@ class OptionalAttachSharedMemory(shared_memory.SharedMemory):
         with self.assertRaises(ValueError):
             sms_invalid = shared_memory.SharedMemory(create=True)
 
+    def test_shared_memory_pickle_unpickle(self):
+        for proto in range(pickle.HIGHEST_PROTOCOL + 1):
+            with self.subTest(proto=proto):
+                sms = shared_memory.SharedMemory(create=True, size=512)
+                self.addCleanup(sms.unlink)
+                sms.buf[0:6] = b'pickle'
+
+                # Test pickling
+                pickled_sms = pickle.dumps(sms, protocol=proto)
+                self.assertNotIn(b'pickle', pickled_sms)
+
+                # Test unpickling
+                sms2 = pickle.loads(pickled_sms)
+                self.assertIsInstance(sms2, sms.__class__)
+                self.assertEqual(sms.name, sms2.name)
+                self.assertEqual(
+                    bytes(sms.buf[0:6]), bytes(sms2.buf[0:6]), b'pickle')
```

Comment:
It perhaps does not work as you expected. The third argument is a message and usually should be string.

Reply:
Yes, that's not how I expected this to work! TIL 👍